BadApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using BadApi.Controllers.Base;
+using BadApi.Domain;
+using BadApi.Models.Post;
+using Dapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace BadApi.Controllers
+{
+	[RoutePrefix("api/post")]
+    public class PostController : BaseApiController
+    {
+		private static List<ViewPostModel> _postCache = new List<ViewPostModel>();
+
+		[HttpGet]
+		[Route("")]
+		public async Task<ViewPostModel> GetPost(long id)
+		{
+			try
+			{
+				if (_postCache.Any(p => p.PostId == id))
+				{
+					return _postCache.First(p => p.PostId == id);
+				}
+
+				using (var connection = GetConnection())
```

Comment:
4 database hits in this connection - consider simplifying into one query using group by